### ISSUE TITLE
[BUG] pillar stack fails to render yaml with tabs in 3006.1

### ISSUE THREAD

**Issue by danielbakken** [saltstack/salt#64320]
**Description**
Salt 3006.1 with ext_pillar stack fails to render Pillar files containing tabs as separators between values and comments. This error does not occur on Salt 3005.1.

**Setup**

/etc/salt/master:
```
ext_pillar:
  - stack: /srv/pillar/config/core.cfg
```

/srv/pillar/config/core.cfg:
```
{%- for role in pillar.get('roles', []) %}
roles/{{ role }}.yml
{%- endfor %}
```

**Steps to Reproduce the behavior**
Configure external pillar stack as described in Setup above. Create /srv/pillar/config/roles/test.yml (replace \t with tab):
```
some_key:
  - some_value\t# some_comment
```

Run `state.test` or `state.apply`:
```
    Data failed to compile:
----------
    Pillar failed to render with the following messages:
----------
    Failed to load ext_pillar stack: Stack pillar yaml parsing error in /srv/pillar/config/roles/test.yml:
while scanning for the next token
found character '\t' that cannot start any token
  in "<unicode string>", line 10, column 15:
      - some_value      # some_comment
                  ^
```

**Expected behavior**

No errors should be returned when running state.apply. This error does not occur in Salt 3005.1.

According to the YAML spec, tab characters are allowed as separators between values and comments, (see below links).

https://yaml.org/spec/1.2.2/#66-comments
Comments must be [separated](https://yaml.org/spec/1.2.2/#separation-spaces) from other tokens by [white space](https://yaml.org/spec/1.2.2/#white-space-characters) characters.

https://yaml.org/spec/1.2.2/#55-white-space-characters
YAML recognizes two white space characters: space and tab.

https://yaml.org/spec/1.2.2/#separation-spaces
Outside [indentation](https://yaml.org/spec/1.2.2/#indentation-spaces) and [scalar content](https://yaml.org/spec/1.2.2/#scalar), YAML uses [white space](https://yaml.org/spec/1.2.2/#white-space-characters) characters for separation between tokens within a line. Note that such [white space](https://yaml.org/spec/1.2.2/#white-space-characters) may safely include [tab](https://yaml.org/spec/1.2.2/#white-space-characters) characters.

**Versions Report**
<details><summary>salt --versions-report</summary>

Salt Version:
          Salt: 3006.1
 
Python Version:
        Python: 3.10.11 (main, May  5 2023, 02:31:54) [GCC 11.2.0]
 
Dependency Versions:
          cffi: 1.14.6
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.0.2
      M2Crypto: 0.38.0
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
  python-gnupg: 0.4.8
        PyYAML: 5.4.1
         PyZMQ: 23.2.0
        relenv: 0.12.3
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
Salt Extensions:
        SSEAPE: 8.11.2.6
 
System Versions:
          dist: rhel 8.7 Ootpa
        locale: utf-8
       machine: x86_64
       release: 4.18.0-425.19.2.el8_7.x86_64
        system: Linux
       version: Red Hat Enterprise Linux 8.7 Ootpa
</details>

**Comment by danielbakken** [comment]
Since this wasn't an issue with previous Salt versions, it might be worth determining what changed in 3006 to cause this regression. If Salt's underlying libraries are to blame, perhaps the maintainers could file a bug report upstream?

Until this is resolved, users must  "fix" previously valid yaml to remove all tabs (which are allowed in the yaml spec) to comply with this undocumented change, or remain on Salt 3005.

I'm happy to help test the yaml library as you request, if you can provide detailed instructions.

**Comment by danielbakken** [comment]
This [pyyaml bug](https://github.com/yaml/pyyaml/issues/450.) appears related, although it mentions trailing tabs and not tabs between values and comments.

I created a test file with a tab (\t):
```
---

some_key:
  - some_value\t#comment
  - value2
```
This same issue occurs with pyyaml 5.4.1 (and also yamllint):

```
>>> import yaml
>>> f = open("test.yml", "r")
>>> yaml.safe_load(f)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3/dist-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "/usr/lib/python3/dist-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/usr/lib/python3/dist-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 82, in compose_node
    node = self.compose_sequence_node(anchor)
  File "/usr/lib/python3/dist-packages/yaml/composer.py", line 110, in compose_sequence_node
    while not self.check_event(SequenceEndEvent):
  File "/usr/lib/python3/dist-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/usr/lib/python3/dist-packages/yaml/parser.py", line 379, in parse_block_sequence_first_entry
    return self.parse_block_sequence_entry()
  File "/usr/lib/python3/dist-packages/yaml/parser.py", line 384, in parse_block_sequence_entry
    if not self.check_token(BlockEntryToken, BlockEndToken):
  File "/usr/lib/python3/dist-packages/yaml/scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "/usr/lib/python3/dist-packages/yaml/scanner.py", line 258, in fetch_more_tokens
    raise ScannerError("while scanning for the next token", None,
yaml.scanner.ScannerError: while scanning for the next token
found character '\t' that cannot start any token
  in "test.yml", line 2, column 15
```

**Comment by Ch3LL** [comment]
I'm actually not able to replicate this with your exact python version and pyyaml library version using the exact copy/paste of your file. But if you are able to replicate the issue outside of Salt this is certainly an upstream issue and would need to be opened on pyyaml's repo.

**Comment by danielbakken** [comment]
Did you replace the "\t" in the test file with a literal tab character?

**Comment by Ch3LL** [comment]
yes: 

```(py-3.10.11)  ch3ll@megan-precision5550  ~/git/pyyaml   master  python --version
Python 3.10.11
(py-3.10.11)  ch3ll@megan-precision5550  ~/git/pyyaml   master  pip list |grep -i pyyaml
PyYAML     5.4.1

[notice] A new release of pip is available: 23.0.1 -> 23.1.2
[notice] To update, run: python3.10 -m pip install --upgrade pip
(py-3.10.11)  ch3ll@megan-precision5550  ~/git/pyyaml   master  cat test.yml 
---

some_key:
  - some_value  #comment
  - value2
(py-3.10.11)  ch3ll@megan-precision5550  ~/git/pyyaml   master  
```

But that's besides the point, if you are able to replicate this issue outside of using Salt, this is an upstream bug. Are you okay closing this here and handling with the upstream project?

**Comment by danielbakken** [comment]
That's fine. I will follow up with pyyaml.
